packages/hooks/useCanPay.ts

```diff
@@ -0,0 +1,22 @@
+import { useBalances } from "./useBalances";
+import { Coin } from "../api/teritori-chain/cosmos/base/v1beta1/coin";
+import { parseUserId } from "../networks";
+
+interface CanPayForPostParams {
+  userId?: string;
+  cost?: Coin;
+}
+
+export const useCanPay = ({ userId, cost }: CanPayForPostParams) => {
+  const [network, userAddress] = parseUserId(userId);
+  const balances = useBalances(network?.id, userAddress);
+  const costBalance = balances.find((bal) => bal.denom === cost?.denom);
+  const canPay =
+    cost &&
+    userId &&
+    !!costBalance?.amount &&
+    parseInt(costBalance.amount, 10) > 0 &&
+    parseInt(costBalance.amount, 10) >= parseInt(cost?.amount || "0", 10);
```

Comment:
`import Long from "long"`

```suggestion
    Long.fromString(costBalance.amount).gt(0) &&
    Long.fromString(costBalance.amount).gte(cost?.amount || "0");
```
some amounts can't be represented with javascript Number

Reply:
Done